.github/scripts/main/main.sh

```diff
@@ -91,6 +91,41 @@ if [[ "$OPAM_DOC" -eq 1 ]]; then
   echo '::endgroup::checking generated files'
 fi
 
+prepare_project () {
+  # warning, perform a cd
+  org=$1
+  project=$2
+
+  url="https://github.com/$org/$project"
+  dir="$CACHE/$project"
+
+  if [ ! -d "$CACHE/$project" ]; then
+    git clone "$url" "$dir"
+  fi
+  cd "$dir"
+  git fetch origin
+  if [ "$GITHUB_EVENT_NAME" = "pull_request" ] && git ls-remote --exit-code origin "$GITHUB_PR_USER/$BRANCH" ; then
+    BRANCH=$GITHUB_PR_USER/$BRANCH
+  fi
+  if git ls-remote --exit-code origin "$BRANCH"; then
+    PR_BRANCH=$BRANCH
+  elif [ "$GITHUB_EVENT_NAME" = pull_request ] && git ls-remote --exit-code origin "$GITHUB_BASE_REF"; then
+    PR_BRANCH=$GITHUB_BASE_REF
+  else
+    PR_BRANCH=master
+  fi
+  if git branch | grep -q "$PR_BRANCH"; then
+    git checkout "$PR_BRANCH"
+    git reset --hard "origin/$PR_BRANCH"
+  else
+    git checkout -b "$PR_BRANCH" "origin/$PR_BRANCH"
+  fi
+
+  test -d _opam || opam switch create . --no-install --formula '"ocaml-system"'
+  eval $(opam env)
```

Comment:
```suggestion
```

Reply:
What is the reason to use `opam exec` instead of `eval opam env` ?

Reply:
eval isn't carried across function boundaries so it would have to be repeated anyway